app/src/main/java/com/example/myapplication/OrderListItemAdapter.java

```diff
@@ -0,0 +1,233 @@
+     *
+     * @param holder
+     */
+    private void expandableLayoutHandler(@NonNull final OrderListItemHolder holder, final OrderListItem model) {
+        holder.infoButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (!holder.expandableView.isExpanded()) {
+                    holder.expandableView.expand(true);
+                } else {
+                    holder.expandableView.collapse(true);
+                }
+            }
+        });
+    }
+
+
+    /**
+     * this function inserts the relevant text to the "order status" textView,
+     * by the relevant order status
+     *
+     * @param holder - the RecyclerView item holder
+     * @param model  - the "Order" object
+     */
+    private void statusTextHandler(OrderListItemHolder holder, OrderListItem model) {
+        if (model.getStatus().equals("open")) {
+                open_order(holder, model);
+        } else if (model.getStatus().equals("locked")) {
+            lock_order(holder);
+        }
+        progressBarHandler(holder, model);
+    }
+
+    /**
+     * a function that change graphics to "locked" status graphics, if order is locked
+     *
+     * @param holder - the RecyclerView item holder
+     */
+    private void lock_order(OrderListItemHolder holder) {
+        holder.statusText.setText("ההזמנה יצאה");
+        holder.joinButton.setText("נעול");
+        holder.joinButton.setBackgroundColor(context.getResources().getColor(R.color.grey));
+        holder.progressBar.setProgressDrawable(context.getDrawable(R.drawable.progress_bar_locked));
+    }
+
+    /**
+     * a function that graphics "open" orders
+     * @param holder - the RecyclerView item holder
+     * @param model - the orderListItem relevant item
+     */
+    private void open_order(OrderListItemHolder holder, OrderListItem model) {
+        holder.joinButton.setText("הצטרף!");
+        holder.joinButton.setBackgroundColor(context.getResources().getColor(R.color.dark_navy));
+        if (model.getPrice() >= CRITICAL_PRICE) {
+            holder.statusText.setText("מוכנה לשילוח!");
+            holder.progressBar.setProgressDrawable(context.getDrawable(R.drawable.progress_bar_green));
+        }
+        else {
+            holder.statusText.setText("מחכה למשבחים...");
+            holder.progressBar.setProgressDrawable(context.getDrawable(R.drawable.progress_bar_orange));
+        }
+
+
+    }
+
+    /**
+     * this function inserts the relevant text to the "order price" textView,
+     * by the relevant order collected money
+     *
+     * @param holder - the RecyclerView item holder
+     * @param model  - the "Order" object
+     */
+    private void priceTextHandler(OrderListItemHolder holder, OrderListItem model) {
+        String priceStr = Integer.toString(model.getPrice());
+        String priceTextInput = String.format("הכסף שנצבר: %s מתוך 70 שקלים", priceStr);
```

Comment:
same here (string resources)

Reply:
Thank you Nir,
Since String Resources connected to Android Activities, and the adapters classes are not 
activities, I believe that a string constant is enough here.

I did what you have asked by using the following manipulation:
Resources.getSystem().getString(R.string.any_string_name);

We can discuss what is a better prcatice .